### PR TITLE
8 patch/api/aticles/articles

### PR DESCRIPTION
**Pull request by alehow84**
Corrections for task 7

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -117,6 +75,94 @@ describe('/api/articles', ()=>{
...
+            .expect(200)
+            .then(({body})=>{
+                expect(body).toBeSortedBy("created_at", {
+                    descending: true
+                })
+            })
+        })
+    })
+    // describe('GET /api/articles invalid requests', ()=>{
+    //     test.skip('400 status responds with "bad request" when given a misspelled endpoint', ()=>{
```
Comment (nc-richard-neat) [inline __tests__/app.test.js]: pls remove comments and console.logs before committing :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -117,6 +75,94 @@ describe('/api/articles', ()=>{
...
+                return request(app)
+                .get('/api/articles/kittens')
+                .expect(400)
+                .then(({body})=>{
+                    expect(body.msg).toBe('bad request')
+                })
+            })
+        })
+        //Q8
+        describe.skip('PATCH /api/articles/:articles_id valid requests', ()=>{
```
Comment (nc-richard-neat) [inline __tests__/app.test.js]: Pls remove any skips or .onlys to allow husky to run all our tests when committing

File: __tests__/app.test.js [hunk 3]
```diff
@@ -166,5 +212,117 @@ describe('/api/articles', ()=>{
...
+        test('201 Adds a new comment object to the database', ()=>{
+            const newComment = {
+                "username": "butter_bridge",
+                "body": "I can't wait for bed-time"
+            }
+            return request(app)
+            .post("/api/articles/5/comments")
+            .send(newComment)
+            .expect(201)
+            .then(() => {
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:302]: Ideally we should be asserting against the response that comes back after posting. If we have a RETURNING * in our post we will get a new comment back which we can assert against 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -166,5 +212,117 @@ describe('/api/articles', ()=>{
...
+                "username": "butter_bridge",
+                "body": "I can't wait for bed-time"
+            }
+            return request(app)
+            .post("/api/articles/5/comments")
+            .send(postedComment)
+            .expect(201)
+            .then(({body})=>{
+                expect(body.postedComment).toMatchObject({
+                    "author": "butter_bridge",
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:322]: I now see you've done the above comment here, we don't need the above assertion then. Also, we will get more properties back on a posted comment so lets assert against all of those properties and values.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -166,5 +212,117 @@ describe('/api/articles', ()=>{
...
+
+an object in the form { inc_votes: newVote }.
+newVote will indicate how much the votes property in the database should be updated by, e.g.
+{ inc_votes : 1 } would increment the current article's vote property by 1
+{ inc_votes : -100 } would decrement the current article's vote property by 100
+Responds with:
+
+the updated article
+Consider what errors could occur with this endpoint, and make sure to test for them.
+
```
Comment (nc-richard-neat) [inline __tests__/app.test.js]: remove comments :)

File: __tests__/app.test.js [hunk 6]
```diff
@@ -166,5 +212,117 @@ describe('/api/articles', ()=>{
...
+                "username": "SleepyGal5000"
+            }
+            return request(app)
+            .post('/api/articles/50/comments')
+            .send(newComment)
+            .expect(404)
+            .then(({body})=>{
+                expect(body.msg).toBe('not found')
+            })
+        })
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:384]: Great error handling!

File: controllers/ncnews.controller.js [hunk 7]
```diff
@@ -47,3 +51,35 @@ exports.getArticleComments = (req, res, next) => {
...
+    const {article_id} = req.params
+
+    //Q7 refactoring to do - articleExistenceQuery below is the start of using Promise.all for advanced error handling
+    // const articleExistenceQuery = checkArticleExists(article_id)  
+
+
+
+    insertArticleComment(newComment, article_id)
+    .then(({rows})=>{
+        const postedComment = rows[0]
```
Comment (nc-richard-neat) [inline controllers/ncnews.controller.js:65]: Convention would be to return rows[0] from the model so no need to access the object here.

**Review by nc-richard-neat — COMMENTED** [review]
Nice one! A few comments to take a look at below but feel free to leave this for another time and move on.

In general try and keep your commits clean by removing all comments and console.logs and only committing code with changes relevant to the current task 👍

File: __tests__/app.test.js [hunk 8]
```diff
@@ -117,6 +75,171 @@ describe('/api/articles', ()=>{
...
+                .expect(400)
+                .then(({body})=>{
+                    expect(body.msg).toBe('bad request')
+                })
+            })
+        })
+        //Q8
+        describe('PATCH /api/articles/:articles_id valid requests', ()=>{
+            test('200 should respond with the article with vote property updated to equal the given vote count where the vote property was previously 0', ()=>{
+                  const newVote = { inc_votes : 1 }
```
Comment (pippaamy) [inline __tests__/app.test.js:144]: I would just send it in your newVote form rather than newVotes

File: __tests__/app.test.js [hunk 9]
```diff
@@ -117,6 +75,171 @@ describe('/api/articles', ()=>{
...
+                const newVotes = {inc_votes: newVote}
+                return request(app)
+                .patch('/api/articles/sukiIs2')
+                .send(newVotes)
+                .expect(400)
+                .then(({body})=>{
+                    expect(body.msg).toBe('bad request')
+                })
+            })
+            //Q8 failing error test
```
Comment (pippaamy) [inline __tests__/app.test.js:230]: if inc_votes is not a number we should get an error - can we test for this?

File: models/ncnews.model.js [hunk 10]
```diff
@@ -38,23 +38,67 @@ exports.fetchAllArticles = () => {
...
+      return db.query(queryStr, params)
+    
+}
+
+exports.amendVotes = (articleId, vote) => {
+    // const voteCount = vote.
+    //define a variable to extract the number in the vote obj
+    //select the articles.vote column where the article_id = articleId from the articles table
+    //insert into articles table a vote count
+    const voteCount = vote.inc_votes.inc_votes
```
Comment (pippaamy) [inline models/ncnews.model.js:88]: if you change how this is sent in the test it doesnt need to do this :)

**Review by pippaamy — COMMENTED** [review]
Great job - I would sent it as just {inc_votes:1} and add that test in :) well done!

**Pull request merged**